sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -207,32 +222,62 @@ case class GlobalLimitExec(limit: Int, child: SparkPlan) extends BaseLimitExec {
  * the child's single output partition.
  */
 case class GlobalLimitAndOffsetExec(
-    limit: Int,
-    offset: Int,
+    limit: Int = -1,
+    offset: Int = 0,
```

Comment:
We don't need a default value for `offset`. If there is no offset, we should use `GlobalLimitExec`.

Reply:
and we can put a `assert(offset > 0)`